PopulateTools/gobierto
# Gobierto People: Person Blog management back-end and UI, and final touches

Connects to #190.

### Review notes

These changes depend on #221.

### What does this PR do?

This PR extends the `GobiertoPeople::Person` data model by adding the Person Post resource, which allows managing Post entries in a Person context. Every Post can have a rich text formatted body field, taking advantage of the `rich_text_area` `FormBuilder` helper (see https://github.com/PopulateTools/gobierto/pull/211). It can be categorised through a list of tags or keywords as well, and they are being stored within the `PersonPost` record. Also, it is providing a simple API to query `PersonPost`s by one or multiple tags.

Finally, this one wraps up some details on the Person resource integration by making People index dynamic, and setting up the deletion strategy.

### How should this be manually tested?

Having `http://gobierto.dev/admin/people/people/<person_id>/blog` (e.g. http://gobierto.dev/admin/people/people/933465536/blog) as the main entry point for this feature, let's check that, after running migrations and re-seeding the database:

- [x] All Person's Posts are listed in the index view.
- [x] A Post can be Added or Edited in this Person's context.
- [x] The Post body field can be managed in rich text format.
- [x] It is also allowed to upload files through the body field.
- [x] The tags field can be managed as a keyword list, separated by comma.

Regarding the final touches on the People view (http://gobierto.dev/admin/people/people):

- [x] All Events, Statements and Posts counters are being calculated dynamically.

---

This is how the UI looks like:

![screen shot 2017-01-09 at 2 43 26 pm](https://cloud.githubusercontent.com/assets/126392/21768481/2f51e588-d67a-11e6-8451-6d3f27e51531.jpg)

---

![screen shot 2017-01-09 at 2 44 09 pm](https://cloud.githubusercontent.com/assets/126392/21768486/33dd6b90-d67a-11e6-88ac-6522392481be.jpg)
